### PR TITLE
Feature/issue 85 clean up permissions

### PR DESCRIPTION
**Pull request by jakeswenson**
Fixes issue #85

### CODE REVIEW

File: core/src/main/java/io/paradoxical/cassieq/admin/resources/api/v1/PermissionsResource.java [hunk 1]
```diff
@@ -99,15 +99,9 @@ public Response generateAuthUrl(@NotNull GetAuthQueryParamsRequest request) thro
...
-        final String computedSignature = signedUrlParameterGenerator.computeSignature(MacProviders.HmacSha256(key));
-
         final String queryParam = SignedUrlParameterNames.queryBuilder()
-                                                         .auth(authorizationLevels)
-                                                         .startTime(request.getStartTime())
-                                                         .endTime(request.getEndTime())
-                                                         .queueName(request.getQueueName())
-                                                         .sig(computedSignature) // always make this last (for prettiness)
-                                                         .build();
+                                                         .fromSignatureGenerator(signedUrlParameterGenerator)
```
Comment (devshorts) [inline core/src/main/java/io/paradoxical/cassieq/admin/resources/api/v1/PermissionsResource.java:103]: Hotttt

**Comment by devshorts** [comment]
I like the changes, though I'm not sure we've really unified signature generator and query Param generator, since both classes have the same set of fields. I was wondering can we use reflection here with some annotations to say how a field is built in the query Param bs how it's made in the signature or something? Then we just add new fields and it auto wires and includes signature values

**Comment by devshorts** [comment]
Either way, still approve the change set. All good improvements.

**Comment by jakeswenson** [comment]
Not sure about the reflection approach, but i get what you're saying. 

I like this approach because a signed query string isn't anything without the signature generator. In this case the SignedUrlSignatureGenerator IS the set of parameters required for query string generation.

**Comment by devshorts** [comment]
Works for me. You're right in that you need the one to complete the other so at least you can't make any bifurcated mistakes there

**Pull request merged**